### ISSUE TITLE
Can this library work with redux-offline?

### ISSUE THREAD

**Issue by larryranches** [Brigad/redux-rest-easy#33]
First off, I really like this library as it simplifies a lot of the redux out of box boilerplate. I'd like to know if this library would work with an "offline" approach specifically with https://github.com/redux-offline/redux-offline?

Is so, are there any code samples?

**Comment by Titozzz** [comment]
Hello, I'm currently using this library with react native and redux offline without major issues, there is still a few things you need to take care of, but theses thing are in our short term todo such as cleaning dead request when re-hydrating the state: 

```
const restEasyTransform = createTransform(
  null,
  outboundState => ({
    requests: {
      ...Object.entries(outboundState.requests || {}).reduce(
        (prev, [key, request]) => ({
          ...prev,
          ...(request.endedAt ? { [key]: request } : {}),
        }),
        {},
      ),
    },
    resources: outboundState.resources,
  }),
  {
    whitelist: ['restEasy'],
  },
);

const customCreateStore = (persistCallback, initialStore = {}) => {
  const customConfig = {
    ...offlineConfig,
    persistCallback,
    persistOptions: {
      whitelist: ['restEasy'],
      transforms: [restEasyTransform],
    },
  };

  return createStore(
    reducers,
    initialStore,
    composeWithDevTools(
      applyMiddleware(thunkMiddleware),
      offline(customConfig),
    ),
  );
};
```

Please note that we are using this in production, so this is one of the feature we will definitely want to support.

We will also add codesandox demos soon.
Please let me know if you have any issue

**Comment by larryranches** [comment]
@Titozzz Thank you for this code sample as this explains a lot. A full demo, I look forward to seeing! Also, is there a roadmap to implement and abstract your solution into a built in utility function with this library?

**Comment by Zephir77167** [comment]
Hey, thanks for your interest in redux-rest-easy!

As @Titozzz pointed out, we want offline to be a first-class citizen, and we plan on providing the state transformer he showed to you directly in the library. Actually, we want a complete version, including:
- pruning of resources not related to any requests anymore (so that outdated resources would not stay in the app forever)
- putting pending requests back to non-pending so that they don't block future requests (the user closed the app while a request was pending) 
- modifying the cache lifetime of requests from `Infinity` to `0` (so they can be used to display an initial state, but would not be trusted forever)

Everything could, of course, end up customizable if some needs we didn't consider were to be pointed out.

Let me know if you have more ideas, I might actually tackle this over the weekend if I have enough time :)

**Comment by Zephir77167** [comment]
Hey, you might be interested in #34 which fixes an issue on first persisted render. I will keep you posted for the transform utility, which is under development.

**Comment by Zephir77167** [comment]
Hey, I just opened #37 which includes the complete utility we were talking about, as well as documentation on how to use it with redux-offline :)

**Comment by larryranches** [comment]
@Zephir77167 Wow that was fast! amazing work

**Comment by Zephir77167** [comment]
Hey, thanks! Additional options we also added in #40 and available since 2.5.0
